source/client-side-encryption/tests/README.rst

```diff
@@ -293,6 +296,81 @@ Then, run the following final tests:
    - Expect an exception to be thrown, since this is an attempt to auto encrypt an already encrypted value.
 
 
+BSON size limits and batch splitting
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, perform the setup.
+
+#. Create a MongoClient without encryption enabled (referred to as ``client``).
+
+#. Using ``client``, drop and create the collection ``db.coll`` configured with the included JSON schema `limits/limits-schema.json <limits/limits-schema.json>`_.
+
+#. Using ``client``, drop the collection ``admin.datakeys``. Insert the document `limits/limits-key.json <limits/limits-key.json>`_
+
+#. Create a MongoClient configured with auto encryption (referred to as ``client_encrypted``)
+
+   Configure with the ``local`` KMS provider as follows:
+
+   .. code:: javascript
+
+      {
+          "local": { "key": <base64 decoding of LOCAL_MASTERKEY> }
+      }
+
+   Where LOCAL_MASTERKEY is the following base64:
+
+   .. code:: javascript
+
+      Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk
+
+   Configure with the ``keyVaultNamespace`` set to ``admin.datakeys``.
+
+Create the following documents:
+
+- ``no_encryption_equal_2mib``
+
+  .. code:: javascript
+
+    {
+        "_id": "no_encryption_equal_2mib",
+        "unencrypted": <the string "a" repeated 2097095 times>
+    }
+
+- ``no_encryption_over_2mib``
+
+  .. code:: javascript
+
+    {
+        "_id": "no_encryption_over_2mib",
+        "unencrypted": <the string "a" repeated 2097097 times>
+    }
+
+- ``encryption_equal_2mib``
+
+  .. code:: javascript
+
+    {
+        "_id": "encryption_equal_2mib",
+        "unencrypted": <the string "a" repeated 2097078 times>,
+        "encrypted": "test"
+    }
+
+
+Using ``client_encrypted`` perform the following operations:
+
+#. Insert ``no_encryption_equal_2mib``. Expect this to succeed.
+
+#. Insert ``no_encryption_over_2mib``. Expect this to throw an exception due to exceeding the reduced maximum BSON document size.
+
+#. Insert ``encryption_equal_2mib``. Expect this to succeed since after encryption this still is below the normal maximum BSON document size.
+
+#. Bulk insert two copies of ``no_encryption_equal_2mib``. Expect the bulk write to succeed and split after first doc (i.e. two inserts occur). This may be verified using `command monitoring <https://github.com/mongodb/specifications/tree/master/source/command-monitoring/command-monitoring.rst>`_.
```

Comment:
Can't insert two copies of this since the _id would be identical

Reply:
Doh, fixed.